Source/WebKit/UIProcess/Cocoa/VideoFullscreenManagerProxy.h

```diff
@@ -257,6 +269,10 @@ class VideoFullscreenManagerProxy : public RefCounted<VideoFullscreenManagerProx
     HashMap<PlaybackSessionContextIdentifier, int> m_clientCounts;
     Vector<CompletionHandler<void()>> m_closeCompletionHandlers;
     WeakHashSet<VideoInPictureInPictureDidChangeObserver> m_pipChangeObservers;
+
+#if HAVE(UI_WINDOW_SCENE_LIVE_RESIZE)
+    Vector<CompletionHandler<void()>> m_playerLayerBoundsHaveChangedCallbacks;
```

Comment:
do you really need a `Vector`? Is there ever more than 1 callback?

Reply:
I don't _think_ so, but I decided to use a Vector to be perfectly safe just in case there does end up being more than 1. Will change though if you think I'm just being paranoid!

Reply:
Well I guess what happens if there are two live resizes which complete before the AV bounds resizes. Does the right thing happen when both callbacks get called or should the callback only get called once?

Reply:
I think the right thing will happen with both callbacks getting called, since each would be associated with its own snapshot view

Reply:
actually, maybe there should only be one callback, and beginLiveResize should call it first to clear it if there is one present?

Reply:
@rr-codes actually I think you could remove all the CompletionHandlers to simplify this.

```
[m_page->cocoaView().liveResizeSnapshotView removeFromSuperview];
```

if you store `liveResizeSnapshotView` in a property in WKWebViewIOS

Reply:
won't the problem of being able to guarantee the snapshot view actually gets removed still be there though?

Reply:
I also need to ensure that it happens strictly after the next presentation update, which I don't think would be possible with this, right?

Reply:
Yes that still needs to be handled. But it solves the problem of making sure the callbacks get called or not.

Reply:
Regarding ensuring it is only removed if needed, you could do something like:
```
[m_page->cocoaView() removeLiveSnapshotIfNeeded];
```

Reply:
This is a minor comment however and I am fine with the callbacks approach. More important is fixing the issue with removing the snapshot when there is no video playing or when the video layer's bounds doesn't actually change after a resize and ensuring the WKWebView's live snapshot is always removed.